core/src/test/scala/org/apache/spark/sql/delta/ExternalPostCommitHookSuite.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.delta.actions.Action
+import org.apache.spark.sql.delta.hooks.PostCommitHook
+import org.apache.spark.sql.delta.test.DeltaSQLCommandTest
+
+class ExternalPostCommitHookSuite extends DeltaGenerateSymlinkManifestSuiteBase
+  with DeltaSQLCommandTest {
+
+  import testImplicits._
+
+  test("generate manifest with postCommitHookClass") {
+    withTempDir { tablePath =>
+      tablePath.delete()
+      withSQLConf(
+        "spark.databricks.delta.properties.defaults.postCommitHookClass" ->
+          classOf[ExternalPostCommitHook].getName) {
+        spark.createDataset(spark.sparkContext.parallelize(1 to 100, 7))
```

Comment:
spark.range(100)?

Reply:
`spark.range(100) ` will not guaranty the number of partitions or task, its depend on the given CPUs. For this test case, the number of partitions should be fixed to determine the number of output files.

Reply:
Makes sense. Why is the number of output files / partition important for this change?

Reply:
Maybe spark.range(100). repartition(7)?